qiskit_nature/second_q/algorithms/ground_state_solvers/ground_state_solver.py

```diff
@@ -19,13 +19,14 @@
 
 from qiskit.opflow import PauliSumOp
 from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info import SparsePauliOp
 
 from qiskit_nature.second_q.operators import SparseLabelOp
 from qiskit_nature.second_q.mappers import QubitConverter, QubitMapper
 from qiskit_nature.second_q.problems import BaseProblem
 from qiskit_nature.second_q.problems import EigenstateResult
 
-QubitOperator = Union[BaseOperator, PauliSumOp]
+QubitOperator = Union[BaseOperator, PauliSumOp, SparsePauliOp]
```

Comment:
I guess this QubitOperator type is the different than the one of the same name in QubitMapper. I had not seen this when I made the prior comment about that type and explicit Union of what it defines elsewhere

Reply:
Actually since `BaseOperator` is already in there, adding `SparsePauliOp` should not be necessary in the first place :thinking: 

But using this `QubitOperator` in the mappers module will not work. So I suppose the path forward will be that in the future:
- the `mappers` module replaces its `QubitOperator` union type with _only_ `SparsePauliOp`
- whereas the `algorithms` module will use _only_ `BaseOperator`

Reply:
We should be careful with a QubitOperator type. This ends up getting exposed to the end user in the API ref and if they mean different things in different places its likely to lead to confusion